src/sql/workbench/browser/modelComponents/table.component.ts

```diff
@@ -116,23 +117,70 @@ export default class TableComponent extends ComponentBase<azdata.TableComponentP
 		}
 	}
 
-	public static transformData(rows: string[][], columns: any[]): { [key: string]: string }[] {
+	private static createIconColumn<T extends Slick.SlickData>(col: azdata.TableColumn): Slick.Column<T> {
+		return <Slick.Column<T>>{
+			name: col.name ?? col.value,
+			id: col.value,
+			field: col.value,
+			width: col.width,
+			cssClass: col.cssClass,
+			headerCssClass: col.headerCssClass,
+			toolTip: col.toolTip,
+			formatter: iconCssFormatter,
+			filterable: false
+		};
+	}
+
+	private static createTextColumn(col: azdata.TableColumn): Slick.Column<any> {
+		return <Slick.Column<any>>{
+			name: col.name ?? col.value,
+			id: col.value,
+			field: col.value,
+			width: col.width,
+			cssClass: col.cssClass,
+			headerCssClass: col.headerCssClass,
+			toolTip: col.toolTip,
+			formatter: textFormatter,
+		};
+	}
+
+
+
+	public static transformData(rows: (string | azdata.IconColumnCellValue)[][], columns: any[], iconCssMap?: { [iconKey: string]: string }): { [key: string]: string | CssIconCellValue }[] {
 		if (rows && columns) {
+			let iconCssCache = iconCssMap ?? [];
 			return rows.map(row => {
-				let object: { [key: string]: string } = {};
-				if (row.forEach) {
-					row.forEach((val, index) => {
-						let columnName: string = (columns[index].value) ? columns[index].value : <string>columns[index];
-						object[columnName] = val;
-					});
+				let object: { [key: string]: string | CssIconCellValue } = {};
+				if (!Array.isArray(row)) {
+					return object;
 				}
+
+				row.forEach((val, index) => {
+					let columnName: string = (columns[index].value) ? columns[index].value : <string>columns[index];
+					if (TableComponent.isIconColumnCellValue(val)) {
+						const icon: IUserFriendlyIcon = val.icon;
+						const iconKey: string = getIconKey(icon);
+						const iconCssClass = iconCssCache[iconKey] ?? createIconCssClass(icon);
+						if (!iconCssCache[iconKey]) {
+							iconCssCache[iconKey] = iconCssClass;
+						}
+
+						object[columnName] = { iconCssClass: iconCssClass, ariaLabel: val.ariaLabel };
+					} else {
+						object[columnName] = <string>val;
+					}
+				});
 				return object;
 			});
 		} else {
 			return [];
 		}
 	}
 
+	private static isIconColumnCellValue(obj: any | undefined): obj is azdata.IconColumnCellValue {
```

Comment:
Put this with the other similar functions in formatters.ts

Reply:
I can't reference azdata interfaces in formatters.ts

Reply:
Duplicating the type isn't really a concern - that's supposed to be the best practice anyways (we already do a ton of that for the other types)

But ok, not a huge deal to keep it here. But please don't have it be a static function - just have it be a top-level function defined in this module outside of the class.

Reply:
ok. I have moved this function out of the class